javascript/0912-sort-an-array.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * https://leetcode.com/problems/sort-an-array/
+ * Merge Sort
+ * Time O(n*log(n)) | Space O(n)
+ * @param {number[]} nums
+ * @return {number[]}
+ */
+var sortArray = function(nums) {
+    
+    const merge = (left, mid, right) => {
+        const arr1 = nums.slice(left, mid+1);
+        const arr2 = nums.slice(mid+1, right + 1);
+        
+        let p1 = 0; 
+        let p2 = 0;
+        let gp = left;
+
+        while(p1 < arr1.length && p2 < arr2.length) {
+            if(arr1[p1] < arr2[p2]) {
+                nums[gp] = arr1[p1];
+                p1++;
+            } else {
+                nums[gp] = arr2[p2];
+                p2++;
+            }
+            gp++;
+        }
+
+        while(p1 < arr1.length) {
+            nums[gp] = arr1[p1];
+            p1++;
+            gp++;
+        }
+
+        while(p2 < arr2.length) {
+            nums[gp] = arr2[p2];
+            p2++;
+            gp++;
+        }
+        return nums;
+    }  
+
+    const mergeSort = (left, right) => {
+
+        if(left === right) return nums;
+        
+        const mid = Math.floor((left+right)/2);
```

Comment:
Suggestion: Use Bitwise
```js
const mid = ((left + right) >> 1 );
```

Reply:
Oh boy. I think it would be too arcane for people to understand. I usually go for this. mid = left + Math.floor((right  - left)/2); But for some reason, I'm not doing that here. Are you sure we should use bit manipulation here? It would be too difficult for people to grasp what we are doing here. That's just a personal opinion. Let me know if you still want me to proceed with the bit manipulation. :)

Reply:
@aadil42  I merged this pr before I got a chance to address this feedback.

The benefit of bitwise when doing binary search is to double dip and allow yourself to build connections of both patterns.

Bitwise is not a common interview problem, but when it is a problem you might forget how it works.
Using it in binary search will reinforce the programming mechanic. Binary search is meant to cut in half, and bitwise can do that.

Reply:
I think I need to think about it clearly. I never halved a number by using the bitwise operator. I will have to look into it. But thanks for the suggestion.